### PR TITLE
Make showRemixer lock across threads, pause ShakeGesture's SensorEventListener when necessary

### PR DESCRIPTION
**Pull request by microfauna**
Addresses #64 and improves #57

### CODE REVIEW

File: remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java [hunk 1]
```diff
@@ -95,17 +99,31 @@ public void onClick(View view) {
...
+    shakeListener = new ShakeListener(activity, threshold, this);
+    shakeListener.attach();
+  }
+
+  /**
+   * Detach from a shake gesture
+   *
+   */
+  public void detachFromShake(final FragmentActivity activity) {
+    shakeListener.detach();
```
Comment (miguelandres) [inline remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java:123]: Let's be extra careful with leaking stuff, though right now this doesn't leak anything (that I can see) add a shakeListener = null; after this.

File: remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java [hunk 2]
```diff
@@ -95,17 +99,31 @@ public void onClick(View view) {
...
-    ShakeListener.attach(activity, threshold, this);
+    shakeListener = new ShakeListener(activity, threshold, this);
+    shakeListener.attach();
+  }
+
+  /**
+   * Detach from a shake gesture
+   *
+   */
+  public void detachFromShake(final FragmentActivity activity) {
```
Comment (miguelandres) [inline remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java]: Remove useless argument

File: remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java [hunk 3]
```diff
@@ -95,17 +99,31 @@ public void onClick(View view) {
...
    */
   public void attachToShake(final FragmentActivity activity, final double threshold) {
-    ShakeListener.attach(activity, threshold, this);
+    shakeListener = new ShakeListener(activity, threshold, this);
+    shakeListener.attach();
+  }
+
+  /**
+   * Detach from a shake gesture
+   *
```
Comment (miguelandres) [inline remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java]: blank line, remove

File: remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java [hunk 4]
```diff
@@ -95,17 +99,31 @@ public void onClick(View view) {
...
+      }
     }
   }
-  
+
+  // TODO(nicksahler): Generalize to attaching to any SensorEventListener
+
   /**
    * Attach this instance to a shake gesture and show fragment when magnitude exceeds {@code threshold}
    *
```
Comment (miguelandres) [inline remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java]: blank line, remove

**Review by miguelandres — CHANGES_REQUESTED** [review]
Minor changes :)

File: remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java (hunk 1 above)
Comment (microfauna) [reply remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java:123]: I thought about this - but it slipped my mind before I pushed. My balance of concerns was the risk of leaking with recreating `shakeListener` on every resume - but risking someone not detaching ever. That said - I will add that for now! 

**update: just realized I am creating a new one on attachToShake. Nevermind - nulling it out.**

File: remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java (hunk 4 above)
Comment (microfauna) [reply remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java]: After TODO line ?

Comment (miguelandres) [reply remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java]: in javadoc, no need to put that extra line there, also you can attach the todo to the Javadoc, or just below it, too.

**Review by miguelandres — APPROVED** [review]

File: remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java [hunk 5]
```diff
@@ -120,10 +118,10 @@ public void attachToShake(final FragmentActivity activity, final double threshol
 
   /**
    * Detach from a shake gesture
-   *
    */
-  public void detachFromShake(final FragmentActivity activity) {
+  public void detachFromShake() {
```
Comment (miguelandres) [inline remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java:122]: update the call in mainActivity!

**Review by miguelandres — CHANGES_REQUESTED** [review]

**Review by miguelandres — APPROVED** [review]

**Pull request merged**